### PR TITLE
Dtype

### PR DESCRIPTION
**Pull request by riveSunder**
Adds dtypes to `v_stability_sweep` and depended-on functions. Also break out `pad_2d` from `ft_convolve`, to allow for padding the convolution kernels ahead of time (instead of re-padding them every time `ft_convolve` is called).

### CODE REVIEW

File: fracatal/functional_jax.py [hunk 1]
```diff
@@ -1,214 +1,224 @@
 from jax import numpy as np
 import numpy.random as npr
 
+def pad_2d(kernel, pad_to):
```
Comment (riveSunder) [inline fracatal/functional_jax.py:4]: Addresses Issue #11

File: fracatal/functional_jax.py [hunk 2]
```diff
@@ -1,214 +1,224 @@
...
+      wm = -1
+
+    padded_kernel = np.pad(kernel, \
+        ((0,0), (0,0), (pad_h+hp, pad_h+hm), (pad_w+wp, pad_w+wm)))
+  else:
+    return kernel
+
+  return padded_kernel
+
+def ft_convolve(grid, kernel, default_dtype=np.float32):
```
Comment (riveSunder) [inline fracatal/functional_jax.py:39]: Related to Issue #12

File: fracatal/functional_jax.py [hunk 3]
```diff
@@ -1,214 +1,224 @@
...
+  else:                                     
+    padded_kernel = kernel                          
+                                        
+  fourier_kernel = np.fft.fft2(np.fft.fftshift(padded_kernel, axes=(-2,-1)), axes=(-2,-1))
+  fourier_grid = np.fft.fft2(np.fft.fftshift(grid, axes=(-2,-1)), axes=(-2,-1))
+  fourier_product = fourier_grid * fourier_kernel 
+  real_spatial_convolved = np.real(np.fft.ifft2(fourier_product, axes=(-2,-1)))
+  convolved = np.fft.ifftshift(real_spatial_convolved, axes=(-2, -1))
+
+  convolved = np.array(convolved, dtype=default_dtype)
```
Comment (riveSunder) [inline fracatal/functional_jax.py:54]: Note that `np.fft` operations unavoidably uses `complex64`!

And yet, `ft_convolve` is still faster than `jax.lax.conv` most of the time!

File: notebooks/fractal_persistence.ipynb [hunk 4]
```diff
@@ -192,7 +192,7 @@
     "    results.append(v_stability_sweep(pattern, make_kernel, my_update, \\\n",
     "            max_t=max_t, max_steps=max_steps, parameter_steps=parameter_steps, stride=stride,\\\n",
     "            min_dt=min_dt, max_dt=max_dt,\\\n",
-    "            min_kr = min_kr, max_kr=max_kr))\n",
+    "            min_kr = min_kr, max_kr=max_kr, default_dtype=np.float16))\n",
```
Comment (riveSunder) [inline notebooks/fractal_persistence.ipynb:195]: Now supports setting dtype, although fft always goes into `complex64`

**Review by riveSunder — COMMENTED** [review]
OK to merge ✓
